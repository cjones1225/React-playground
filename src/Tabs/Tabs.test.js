import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Tabs from "./Tabs";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

describe(`Tabs Component`, () => {
  const tabsProp = [
    { name: "First Tab", content: "Lorem Ipsum dolor sit amet" },
    { name: "Second Tab", content: "Laboriosam exercitationem" },
    { name: "Third Tab", content: "Fugit, sapiente aspernatur" }
  ];
  it("renders without erros", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Tabs />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the first tab by default", () => {
    const tree = renderer.create(<Tabs tabs={tabsProp} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders empty given no tabs", () => {
    const wrapper = shallow(<Tabs />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
