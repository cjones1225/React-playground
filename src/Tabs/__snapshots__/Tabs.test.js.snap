// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component renders empty given no tabs 1`] = `<div />`;

exports[`Tabs Component renders the first tab by default 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    First Tab
  </button>
  <button
    onClick={[Function]}
  >
    Second Tab
  </button>
  <button
    onClick={[Function]}
  >
    Third Tab
  </button>
  <div
    className="content"
  >
    Lorem Ipsum dolor sit amet
  </div>
</div>
`;
